djangocms_text_ckeditor/static/djangocms_text_ckeditor/ckeditor_plugins/cmsplugins/plugin.js

```diff
@@ -21,105 +43,134 @@ $(document).ready(function () {
+                onBlock: function (panel, block) {
                     block.element.setHtml(that.editor.plugins.cmsplugins.setupDropdown());
 
                     var anchors = $(block.element.$).find('.cke_panel_listItem a');
-                        anchors.bind('click', function (e) {
-                            e.preventDefault();
 
-                            that.addPlugin($(this), panel);
-                        });
+                    anchors.bind('click', function (e) {
+                        e.preventDefault();
+
+                        that.addPlugin($(this), panel);
+                    });
                 }
             });
 
             // handle edit event via context menu
-            if(this.editor.contextMenu) {
+            if (this.editor.contextMenu) {
                 this.setupContextMenu();
-                this.editor.addCommand('cmspluginsEdit', {
-                    exec: function () {
-                        var selection = that.editor.getSelection();
-                        var element = selection.getSelectedElement() || selection.getCommonAncestor().getAscendant('a', true);
-                        that.editPlugin(element);
-                    }
-                });
             }
 
+            this.editor.addCommand('cmspluginsEdit', {
+                exec: function () {
+                    var element = that.getElementFromSelection();
+                    var plugin = that.getPluginWidget(element);
+
+                    if (plugin) {
+                        that.editPlugin(plugin);
+                    }
+                }
+            });
+
             // handle edit event on double click
             // if event is a jQuery event (touchend), than we mutate
             // event a bit so we make the payload similar to what ckeditor.event produces
-            var handleEdit = function(event) {
+            var handleEdit = function (event) {
+                var element;
+
                 if (event.type === 'touchend' || event.type === 'click') {
-                    var element = event.currentTarget;
-                    event.data = event.data ||  {};
-                    that.editor.getSelection().fake(new CKEDITOR.dom.element(element));
-                } else {
-                    // heavily relies on the fact that double click
-                    // also selects an element
-                    var selection = that.editor.getSelection();
-                    var element = selection.getSelectedElement() || selection.getCommonAncestor().getAscendant('a', true);
-                }
-                if(element && element.getAttribute('id').indexOf('plugin_obj_') === 0) {
-                    event.data.dialog = '';
-                    that.editPlugin(element);
+                    // pick cke_widget span
+                    // eslint-disable-next-line new-cap
+                    element = new CKEDITOR.dom.element($(event.currentTarget).closest('cms-plugin')[0]).getParent();
```

Comment:
shorter line?
`var plugin = $(event.currentTarget).closest('cms-plugin')[0];`
`element = new CKEDITOR.dom.element(plugin).getParent();`

also you do not check if that plugin element exist